### PR TITLE
Deploy upcoming

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: code/model/DNEnvironment.php [hunk 1]
```diff
@@ -614,13 +614,24 @@ public function CurrentBuild() {
...
-	 * A history of all builds deployed to this environment
-	 *
+	 * A list of past deployments.
 	 * @return ArrayList
 	 */
 	public function DeployHistory() {
 		return $this->Deployments()
-			->where('SHA IS NOT NULL')
+			->where('"SHA" IS NOT NULL')
+			->filter('State', ['Completed', 'Failed', 'Invalid'])
```
Comment (stojg) [inline code/model/DNEnvironment.php]: Can you use the state class const defined?

File: code/model/DNEnvironment.php [hunk 2]
```diff
@@ -614,13 +614,24 @@ public function CurrentBuild() {
...
+	}
+
+	/**
+	 * A list of upcoming or current deployments.
+	 * @return ArrayList
+	 */
+	public function UpcomingDeployments() {
+		return $this->Deployments()
+			->where('"SHA" IS NOT NULL')
+			->filter('State', ['New', 'Submitted', 'Deploying', 'Aborting'])
```
Comment (stojg) [inline code/model/DNEnvironment.php]: Can you use the state class const defined?

File: code/model/DNEnvironment.php (hunk 1 above)
Comment (halkyon) [inline code/model/DNEnvironment.php]: so you want the states defined as an array somewhere? You really love your constants don't you :D

Comment (stojg) [inline code/model/DNEnvironment.php]: I just dislike magic numbers and strings :) I have fat fingers and it's easy to miss uppercase / lowercase as well. :P

File: code/model/DNEnvironment.php (hunk 2 above)
Comment (halkyon) [inline code/model/DNEnvironment.php]: fixed

File: code/model/DNEnvironment.php (hunk 1 above)
Comment (halkyon) [inline code/model/DNEnvironment.php]: fixed

**Pull request merged**